### PR TITLE
Spring cleaning fall 2017

### PR DESCRIPTION
**Pull request by djmitche**
I'm a little nervous about the azure-storage changes, but the tests seem pretty thorough.

Advice on how to better test the changes before deploying is appreciated :)

### CODE REVIEW

File: test/artifact_test.js [hunk 1]
```diff
@@ -5,14 +5,13 @@ suite('Artifacts', function() {
...
-  var Promise       = require('promise');
-  var request       = require('superagent-promise');
+  var request       = require('superagent');
   var assert        = require('assert');
   var urljoin       = require('url-join');
   var BlobUploader  = require('./azure-blob-uploader-sas');
-  var Bucket        = require('../lib/bucket');
-  var BlobStore     = require('../lib/blobstore');
-  var data          = require('../lib/data');
+  var Bucket        = require('../src/bucket');
```
Comment (jonasfj) [inline test/artifact_test.js]: isn't this just `./` instead of `../`

Comment (djmitche) [reply test/artifact_test.js]: The paths are relative to the source file, `tests/artifact_test.js`.  So `./src` wouldn't be found.

**Comment by djmitche** [comment]
I'd like review from @imbstack and @jhford as well, the latter especially for the azure changes.

**Comment by djmitche** [comment]
Rebased on master with conflicts resolved..

**Comment by djmitche** [comment]
I'll land this tomorrow

**Pull request merged**